lib/ansible/modules/system/parted.py

```diff
@@ -0,0 +1,654 @@
+        'model': generic_params[6],
+        'logical_block':  int(generic_params[3]),
+        'physical_block': int(generic_params[4])
+    }
+
+    parts = []
+    for line in lines[2:]:
+        line = line.strip().rstrip(';')
+        if not line:
+            continue
+
+        part_params = line.split(':')
+        flags = [f for f in part_params[6].split(', ') if f != '']
+
+        parts.append({
+            'num':    int(part_params[0]),
+            'begin':  parse_unit(part_params[1], unit)[0],
+            'end':    parse_unit(part_params[2], unit)[0],
+            'size':   parse_unit(part_params[3], unit)[0],
+            'fstype': part_params[4] or None,
+            'flags':  flags,
+        })
+
+    return {'generic': generic, 'partitions': parts}
+
+
+def format_disk_size(size_bytes, unit):
+    """
+    Formats a size in bytes into a different unit, like parted does. It doesn't
+    manage CYL and CHS formats, though.
+    This function has been adapted from https://github.com/Distrotech/parted/blo
+    b/279d9d869ff472c52b9ec2e180d568f0c99e30b0/libparted/unit.c
+    """
+    unit = unit.lower()
+
+    # Shortcut
+    if size_bytes == 0:
+        return 0.0
+
+    # Reference prefixes (International System of Units and IEC)
+    units_si  = ['b', 'kb',  'mb',  'gb',  'tb',  'pb',  'eb',  'zb',  'yb']
+    units_iec = ['b', 'kib', 'mib', 'gib', 'tib', 'pib', 'eib', 'zib', 'yib']
+
+    # Cases where we default to 'compact'
+    if unit in ['', 'compact', 'cyl', 'chs']:
+        index = max(0, int(
+            (math.log10(size_bytes) - 1.0) / 3.0
+        ))
+        unit = 'b'
+        if index < len(units_si):
+            unit = units_si[index]
+
+    # Find the appropriate multiplier
+    multiplier = 1.0
+    if unit in units_si:
+        multiplier = 1000.0 ** units_si.index(unit)
+    elif unit in units_iec:
+        multiplier = 1024.0 ** units_iec.index(unit)
+
+    output = size_bytes / multiplier * (1 + 1E-16)
+
+    # Corrections to round up as per IEEE754 standard
+    if output < 10:
+        w = output + 0.005
+    elif output < 100:
+        w = output + 0.05
+    else:
+        w = output + 0.5
+
+    if w < 10:
+        precision = 2
+    elif w < 100:
+        precision = 1
+    else:
+        precision = 0
+
+    # Round and return
+    return round(output, precision), unit
+
+
+def read_record(file_path, default=None):
+    """
+    Reads the first line of a file and returns it.
+    """
+    try:
+        f = open(file_path, 'r')
+        try:
+            return f.readline().strip()
+        finally:
+            f.close()
+    except IOError:
+        return default
+
+
+def get_unlabeled_device_info(device, unit):
+    """
+    Fetches device information directly from the kernel and it is used when
+    parted cannot work because of a missing label. It always returns a 'unknown'
+    label.
+    """
+    device_name = os.path.basename(device)
+    base = "/sys/block/{0}".format(device_name)
```

Comment:
Using format() here means it won't work with Python 2.6 and older. Since Ansible supports Python 2.4 and higher you should really be using the old formatting method.
See http://pyformat.info/